config/clusters/eks.tf

```diff
@@ -40,29 +40,62 @@ module "eks" {
       kubelet_extra_args = "--kube-reserved=emephemeral-storage=30Gi"
       k8s_labels = {
         Archtype    = "x86"
+        Application = "prow"
         Environment = "training"
         GithubRepo  = "terraform-aws-eks"
         GithubOrg   = "terraform-aws-modules"
       }
-      additional_tags = {
-        ExtraTag = "falco"
+    }
+
+    jobs = {
+      name             = var.eks_jobs_worker_group_name
+      desired_capacity = var.eks_jobs_worker_group_asg_desired_capacity
+      max_capacity     = var.eks_jobs_worker_group_asg_max_capacity
+      min_capacity     = var.eks_jobs_worker_group_asg_min_capacity
+
+      # This Node Group is available in a Single Availability Zone.
+      # This should avoid AutoScaling to conflict with cluster-autoscaler
+      # during intervention after AZ-rebalance.
+      # Read more here: https://github.com/kubernetes/autoscaler/issues/3693
+      # This is needed to guarantee QoS for long-running build jobs.
+      subnets = [module.vpc.private_subnets[0]]
+      taints  = [local.single_az_nodegroup_taint]
+
+      instance_types     = [var.eks_jobs_worker_group_instance_type]
+      kubelet_extra_args = "--kube-reserved=emephemeral-storage=30Gi"
+      ami_type           = "AL2_x86_64"
+      version            = var.eks_cluster_version
+
+      k8s_labels = {
+        Archtype    = "x86"
+        Application = "jobs"
+        Environment = "training"
+        GithubRepo  = "terraform-aws-eks"
+        GithubOrg   = "terraform-aws-modules"
       }
     }
-    arm-ng = {
-      name_prefix        = null
-      desired_capacity   = var.eks_arm_worker_group_asg_desired_capacity
-      max_capacity       = var.eks_arm_worker_group_asg_max_capacity
-      min_capacity       = var.eks_arm_worker_group_asg_min_capacity
-      instance_types     = ["m6g.large"]
-      ami_type           = "AL2_ARM_64"
+
+    jobs_arm = {
+      name             = var.eks_jobs_arm_worker_group_name
+      desired_capacity = var.eks_jobs_arm_worker_group_asg_desired_capacity
+      max_capacity     = var.eks_jobs_arm_worker_group_asg_max_capacity
+      min_capacity     = var.eks_jobs_arm_worker_group_asg_min_capacity
+      instance_types   = [var.eks_jobs_arm_worker_group_instance_type]
+
+      ami_type = "AL2_ARM_64"
+      taints   = [local.arm_nodegroup_taint]
```

Comment:
Shouldn't this have also the `local.single_az_nodegroup_taint`?

Reply:
Nope @FedeDP. It's deployed in multi-AZ here

Reply:
Mmh so build drivers arm64 jobs will still risk to be evicted?

Reply:
I didn't see the amount of eviction for arm builds right now. But we can do it also for them. WDYT?

Reply:
I can confirm that we experienced some evictions also from arm nodes in one of the last runs, so maybe it's a good idea to add the taint

Reply:
Applied @FedeDP @cappellinsamuele :)